src/core/Akka/Actor/ActorRef.cs

```diff
@@ -121,43 +104,37 @@ public override IActorRefProvider Provider
         /// <param name="sender">TBD</param>
         protected override void TellInternal(object message, IActorRef sender)
         {
+            var handled = false;
 
-            if (message is ISystemMessage sysM) //we have special handling for system messages
-            {
-                SendSystemMessage(sysM);
-            }
-            else
+            switch (message)
             {
-                if (Interlocked.Exchange(ref status, COMPLETED) == INITIATED)
```

Comment:
Do we lose thread-safety without this? What happens if multiple threads try to complete the same Ask at the same time?